lib/firebase.dart

```diff
@@ -72,3 +74,32 @@ void _fbAuthChanged(
 
   loginCallback();
 }
+
+// Read data
+Future<model.Config> fetchConfig() async {
+  logger.debug('Fetching config from firebase ..');
+  var chartsConfigRef = firebase.firestore().doc(fb_constants.metadataPath);
+  var configSnap = await chartsConfigRef.get();
+  var configData = configSnap.data();
+
+  return model.Config.fromData(configData);
+}
+
+Future<Map<String, Map<String, dynamic>>> fetchInteractions(String path) async {
+  if (path == null || path == '') {
+    throw Error();
```

Comment:
nit: prefer to use a more specific error, and supply an error message as well, e.g. something like:
```suggestion
    throw ArgumentError("Path for fetching interactions cannot be empty or null");
```

Reply:
Ah yes I was searching for how to create a class of errors. Came up with Exceptions, but that didn't look right - so passed a generic error. Fixed